### PR TITLE
DRY up logic slightly

### PR DESCRIPTION
**Pull request by ErikSchierboom**

### CODE REVIEW

File: app/models/git/exercise.rb [hunk 1]
```diff
@@ -162,13 +162,15 @@ def tooling_files
       end
     end
 
-    # This includes meta files
+    # This includes meta files, but excludes docs files
     memoize
     def tooling_filepaths
-      filepaths
+      filepaths.reject do |filepath|
+        filepath.starts_with?(".docs")
```
Comment (iHiD) [inline app/models/git/exercise.rb]: Worth making this `.docs/`?

**Comment by iHiD** [comment]
This looks good. Worth quickly double checking with `#maintainers` that none of them are using `.docs` for anything?

File: app/models/git/exercise.rb (hunk 1 above)
Comment (ErikSchierboom) [reply app/models/git/exercise.rb]: Absolutely.

File: app/commands/submission/representation/init.rb [hunk 2]
```diff
@@ -28,7 +28,7 @@ def call
 
       def exercise_filepaths
         exercise_repo.tooling_filepaths.select do |filepath|
-          filepath.starts_with?(".meta")
+          filepath.starts_with?(".meta/")
```
Comment (ErikSchierboom) [inline app/commands/submission/representation/init.rb]: This is the only real exception, where we currently don't output the test/editor files for the representer.

File: app/commands/submission/test_run/init.rb [hunk 3]
```diff
@@ -43,15 +43,8 @@ def call
...
-        exercise_repo.tooling_filepaths.select do |filepath|
-          # Skip non-functional files
-          next false if filepath.starts_with?(".docs")
-          next false if filepath == "README.md"
-
-          # Skip submitted files
-          next false if submission.valid_filepaths.include?(filepath)
-
-          true
+        exercise_repo.tooling_filepaths.reject do |filepath|
```
Comment (ErikSchierboom) [inline app/commands/submission/test_run/init.rb]: I've tried moving this functionality to submission, as it is the same here and for the analyzer, but didn't do that as the `exercise_repo` used here is not necessarily the same as the submission exercise's git repo (due to the test run being allowed to pass in the `git_sha`).

File: test/commands/submission/analysis/init_test.rb [hunk 4]
```diff
@@ -22,7 +22,36 @@ class Submission::Analysis::InitTest < ActiveSupport::TestCase
...
         exercise_git_dir: exercise_repo.dir,
-        exercise_filepaths: [".meta/config.json", ".meta/design.md", ".meta/exemplar.rb"]
+        exercise_filepaths: [".meta/config.json", ".meta/design.md", ".meta/exemplar.rb", "log_line_parser_test.rb"]
+      }
+    )
+    Submission::Analysis::Init.(submission)
+    assert_equal 'queued', submission.analysis_status
+  end
+
+  test "calls to publish_message for exercise with editor files" do
```
Comment (ErikSchierboom) [inline test/commands/submission/analysis/init_test.rb:33]: This test case verifies that the original issue (editor files not being present)

**Comment by iHiD** [comment]
@ErikSchierboom Any risks in merging this?

**Comment by ErikSchierboom** [comment]
@iHiD This is technically a breaking change, as tooling no longer receives the `.docs` files. However, I asked in the #maintainers channel if people used files in the `.docs` folder in their tooling, and they told me they didn't. I also find it unlikely they do.

What this _does_ do is to give tooling access to editor files, which is something that does not happen at the moment.

**Comment by iHiD** [comment]
Happy for this to go down, but think #2348 might be worth merging into this first :)

**Comment by ErikSchierboom** [comment]
@iHiD Merged. Those changes look good.

**Pull request merged**